packages/admin/src/Charcoal/Admin/Action/ElfinderConnectorAction.php

```diff
@@ -186,6 +190,18 @@ public function setupElfinder(array $extraOptions = [])
             define('ELFINDER_IMG_PARENT_URL', (string)$this->baseUrl(ElfinderTemplate::ELFINDER_ASSETS_REL_PATH));
         }
 
+        $extraOptions = array_merge($extraOptions, ['bind' => [
+            'upload.presave' => [function (&$thash, &$name, $src) {
+                if (!$src || !file_exists($src)) {
+                    return false;
+                }
+
+                $this->getEventDispatcher()->dispatch(new FileWasUploaded($src));
+
+                return true;
+            }]
+        ]]);
```

Comment:
Improve coding style of multi-line array.

```suggestion
        $extraOptions = array_merge($extraOptions, [
            'bind' => [
                'upload.presave' => [
                    function (&$thash, &$name, $src) {
                        if (!$src || !file_exists($src)) {
                            return false;
                        }

                        $this->getEventDispatcher()->dispatch(new FileWasUploaded($src));

                        return true;
                    },
                ],
            ],
        ]);
```

Reply:
Move anonymous function to a method for easier customization.

```suggestion
        $extraOptions = array_merge($extraOptions, [
            'bind' => [
                'upload.presave' => [
                    ":dispatchEventOnUploadPreSave",
                ],
            ],
        ]);
```

```php
    /**
     * Dispatches an event on `upload.presave`.
     *
     * @param  string $path     The target path.
     * @param  string $name     The target name.
     * @param  string $src      The temporary file name.
     * @param  object $elfinder The elFinder instance.
     * @param  object $volume   The current volume instance.
     * @return void|bool|array
     */
    public function dispatchEventOnUploadPreSave(&$path, &$name, $src, $elfinder, $volume)
    {
        if (!$src || !file_exists($src)) {
            return false;
        }

        $this->getEventDispatcher()->dispatch(new FileWasUploaded($src));

        return true;
    }
```